src/components/cleaner.ts

```diff
@@ -45,48 +45,32 @@ export class Cleaner {
-            globs.map(g => this.globP(g, {cwd: outdir}))
-        ).then(files => files
-            // Reduce the array of arrays to a single array containing all the files that should be deleted
-            .reduce((all, curr) => all.concat(curr), [])
-            // Resolve the absolute filepath for every file
-            .map(file => path.resolve(outdir, file))
-        ).then(files => Promise.all(
-            // Try to unlink the files, returning a Promise for every file
-            files.map(file => fs.promises.unlink(file).then(() => {
-                this.extension.logger.addLogMessage(`File cleaned: ${file}`)
-                // If unlinking fails, replace it with an rmdir Promise
-            }, () => fs.promises.rmdir(file).then(() => {
-                this.extension.logger.addLogMessage(`Folder removed: ${file}`)
-            }, () => {
-                this.extension.logger.addLogMessage(`Error removing file: ${file}`)
-            })))
-        )).then(
-            () => {} // Do not pass results to Promise returned by clean()
-        ).catch(err => {
-            this.extension.logger.addLogMessage(`Error during deletion of files: ${err}`)
-            if (err instanceof Error) {
-                this.extension.logger.logError(err)
-            }
-        })
-    }
-
-    // This function wraps the glob package into a promise.
-    // It behaves like the original apart from returning a Promise instead of requiring a Callback.
-    private globP(pattern: string, options: glob.IOptions): Promise<string[]> {
-        return new Promise((resolve, reject) => {
-            glob(pattern, options, (err, files) => {
-                if (err) {
-                    reject(err)
+        return new Promise<void>((resolve, _reject) => {
+            const files = globs.map(g => glob.sync(g, {cwd: outdir}))
+                // Reduce the array of arrays to a single array containing all the files that should be deleted
+                .reduce((all, curr) => all.concat(curr), [])
+                // Resolve the absolute filepath for every file
+                .map(file => path.resolve(outdir, file))
+            for (const file of files) {
+                const stats: fs.Stats = fs.statSync(file)
+                if (stats.isFile()) {
+                    fs.unlink(file, err => {
+                        if (!err) {
+                            this.extension.logger.addLogMessage(`Cleaning file: ${file}`)
+                        } else {
+                            this.extension.logger.addLogMessage(`Error while cleaning: ${file}`)
+                            this.extension.logger.addLogMessage(err.message)
+                        }
+                    })
                 } else {
-                    resolve(files)
+                    this.extension.logger.addLogMessage(`Not removing non-file: ${file}`)
                 }
-            })
+            }
+            resolve()
```

Comment:
The code is not correct. `fs.unlink` asynchronously removes a file. When `resolve()` called, the removal might not finish yet. 

- https://nodejs.org/docs/latest-v14.x/api/fs.html#fs_fs_unlink_path_callback

We can use `await`.

```ts
    private async cleanGlob(rootFile: string): Promise<void> {
        const configuration = vscode.workspace.getConfiguration('latex-workshop')
        let globs = configuration.get('latex.clean.fileTypes') as string[]
        const outdir = path.resolve(path.dirname(rootFile), this.extension.manager.getOutDir(rootFile))
        if (configuration.get('latex.clean.subfolder.enabled') as boolean) {
            globs = globs.map(globType => './**/' + globType)
        }
        
        const files = globs.map(g => glob.sync(g, {cwd: outdir}))
                           // Reduce the array of arrays to a single array containing all the files that should be deleted
                           .reduce((all, curr) => all.concat(curr), [])
                           // Resolve the absolute filepath for every file
                           .map(file => path.resolve(outdir, file))
        for (const file of files) {
            const stats: fs.Stats = fs.statSync(file)
            if (stats.isFile()) {
                try {
                    await fs.promises.unlink(file)
                } catch (err) {
                    this.extension.logger.addLogMessage(`Error while cleaning: ${file}`)
                    this.extension.logger.addLogMessage(err.message)
                    continue
                }
                this.extension.logger.addLogMessage(`Cleaning file: ${file}`)
            } else {
                this.extension.logger.addLogMessage(`Not removing non-file: ${file}`)
            }
        }
    }
```